### DISCUSSION TITLE
ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error"

### DISCUSSION THREAD

**Discussion by NVG2021** [Ride-The-Lightning/RTL#809] [Troubleshooting Questions] [answered]
Hello, I've got this error on RTL terminal screen after installing
Bitcoincore
c-lightning 
Node
C-lighnting-Rest
RTL
[2021-09-14T13:49:06.689Z] ERROR: GetInfo => Get Info Error: {"name":"RequestError","message":"Error: connect ECONNREFUSED 127.0.0.1:8080","cause":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"options":{"url":"https://localhost:8080/v1/getinfo","rejectUnauthorized":false,"json":true,"headers":{},"method":"GET","qs":{},"simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false}}

I've been trying to see where I introduced this 127.0.0.1:8080 ... but I do not remember to have introduced it, unless it comes by default on any install procces or sample config file....

RTL does not see the c-lightning node nor shows any info about the local node... which is running and with peers on sight...
What can be the cause???

Many thanks for the support (as newbeeeee)
nicolas 

**To Reproduce** Open RTL from browser on http://localhost:3010

**Screenshots**

**Your environment**
* Version of `RTL` -  0.11.0-beta
* Version of `lnd`- I have c-lightning - v0.10.1-16-g4514d2a
* Version of `btcd`, `bitcoind`, or other backend -  0.21.1
* Browser & browser versionFirefox
* Operating system (`uname -a` on *Nix) Linux Ubuntu 20.0444
* any other relevant environment details

**Additional context**

**Comment by NVG2021** [comment]
Hello, I've found that I've used a wrong port, so I used the 3002, which is the correct one for the C-lightning and now I get the following new error after loging in into RTL .... and it also shows on the 

Error: write eproto 140042077263680:error:1408f10b:ssl routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:331: 

I am running on node version v10.19.0

**Comment by saubyk** [comment]
Hi @NVG2021 
- What's the port configured for RTL in the `RTL-Config.json` file?
- What's the port configured for C-Lightning-REST?
- Have you configured the C-Lightning-REST macaroon in the RTL-Config file?

Also your node version is quite old, you should upgrade to at least v14

**Comment by NVG2021** [comment]
Hi, many thanks for the reply,
find the info of the files,
But I fixed part and now there is anew error message as I stated on a new comment there

RTL-Config.json
{
  "port": "3010",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "NironetNode",
      "lnImplementation": "CLT",
      "Authentication": {
        "macaroonPath": "/home/nicolas/c-lightning-REST/certs/",
        "configPath": "/home/nicolas/.lightning/config"
      },
      "Settings": {
        "userPersona": "MERCHANT",
        "themeMode": "DAY",
        "themeColor": "INDIGO",
        "channelBackupPath": "/home/nicolas/backup/node-1",
        "logLevel": "ERROR",
        "lnServerUrl": https://localhost:3001/v1,
        "fiatConversion": true,
        "currencyUnit": "USD"
      }
    }
The the password hash

  GNU nano 4.8                                                  cl-rest-config.json
{
    "PORT": 3001,
    "DOCPORT": 4001,
    "PROTOCOL": "https",
    "EXECMODE": "production",
    "RPCCOMMANDS": ["*"]
}


  *   Have you configured the C-Lightning-REST macaroon in the RTL-Config file? –
I guess so, and the macaroon file is the given place… don’t know if I should end the  path with “/” or without
         "macaroonPath": "/home/nicolas/c-lightning-REST/certs/", OR         "macaroonPath": "/home/nicolas/c-lightning-REST/certs",





Nicolas Velasco
Gestion – Formación Empresas - Energía

***@***.***
http://www.nironet.es<http://www.nironet.es/>
[twitter]<https://twitter.com/NironetSchool>[sÃ­mbolos-redes-sociales-02-1]<https://www.facebook.com/Nironet>
Nironet SL
Barcelona – Vilanova i la Geltrú
Tel: 938 106 596
Movil: (+34) 607 403 073


De: Suheb ***@***.***>
Enviado el: martes, 14 de septiembre de 2021 17:43
Para: Ride-The-Lightning/RTL ***@***.***>
CC: Nicolas Velasco - NR ***@***.***>; Mention ***@***.***>
Asunto: Re: [Ride-The-Lightning/RTL] ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error" (#808)


Hi @NVG2021<https://github.com/NVG2021>

  *   What's the port configured for RTL in the RTL-Config.json file?
  *   What's the port configured for C-Lightning-REST?
  *   Have you configured the C-Lightning-REST macaroon in the RTL-Config file?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Ride-The-Lightning/RTL/issues/808#issuecomment-919274270>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVUAE6B4NSD6HZ4GVTZMCKLUB5UQHANCNFSM5EAHLV3A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
[Atención: E-mail Externo]
//--------------------------------------------------------------------// Por favor, revise nuestra política de protección de datos: Per favor, revisi la nostra política de protecció de dades: https://www.nironet.es/politicas-de-cookies-y-proteccion-de-datos-2 * Antes de imprimir este e-mail, piense en si es realmente necesario: El Medio Ambiente es responsabilidad de todos. * Abans d'imprimir aquest e-mail, pensi en si és realment necessari: El Medi ambient és responsabilitat de tots. * Before printing this e-mail, think if you really need to so. The environment is everyone's responsibility. //--------------------------------------------------------------------//

**Comment by saubyk** [comment, marked as answer]
Please update this line in the RTL-Config.json file...
`"lnServerUrl": https://localhost:3001/v1,`
to
`"lnServerUrl": "https://localhost:3001/v1",`

Also, please upgrade your node version to at least v14.

**Comment by NVG2021** [comment]
Hi, thanks.. will do this night as now I have to go… will let u know the results

Many thanks

Nicolas Velasco
Gestion – Formación Empresas - Energía

***@***.***
http://www.nironet.es<http://www.nironet.es/>
[twitter]<https://twitter.com/NironetSchool>[sÃ­mbolos-redes-sociales-02-1]<https://www.facebook.com/Nironet>
Nironet SL
Barcelona – Vilanova i la Geltrú
Tel: 938 106 596
Movil: (+34) 607 403 073


De: Suheb ***@***.***>
Enviado el: martes, 14 de septiembre de 2021 18:07
Para: Ride-The-Lightning/RTL ***@***.***>
CC: Nicolas Velasco - NR ***@***.***>; Mention ***@***.***>
Asunto: Re: [Ride-The-Lightning/RTL] ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error" (#808)


Please update this time in the RTL-Config.json file...
"lnServerUrl": https://localhost:3001/v1,
to
"lnServerUrl": "https://localhost:3001/v1",

Also, please upgrade your node version to at least v14.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Ride-The-Lightning/RTL/issues/808#issuecomment-919295511>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVUAE6EWOJQCRTSDFUXINV3UB5XJFANCNFSM5EAHLV3A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
[Atención: E-mail Externo]
//--------------------------------------------------------------------// Por favor, revise nuestra política de protección de datos: Per favor, revisi la nostra política de protecció de dades: https://www.nironet.es/politicas-de-cookies-y-proteccion-de-datos-2 * Antes de imprimir este e-mail, piense en si es realmente necesario: El Medio Ambiente es responsabilidad de todos. * Abans d'imprimir aquest e-mail, pensi en si és realment necessari: El Medi ambient és responsabilitat de tots. * Before printing this e-mail, think if you really need to so. The environment is everyone's responsibility. //--------------------------------------------------------------------//

**Comment by NVG2021** [comment]
Hi there,

I’ve  upgraded to node version:


And now the error is as follows
On terminal
[2021-09-14T21:09:31.798Z] ERROR: GetInfo => Get Info Error: {"name":"RequestError","message":"Error: connect ECONNREFUSED 127.0.0.1:3001","cause":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"error":{"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"options":{"url":https://localhost:3001/v1/getinfo,"rejectUnauthorized":false,"json":true,"headers":{},"method":"GET","qs":{},"simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false}}

On RTL webapp
***@***.******@***.***

The config files have the following text:

RTL-Config.json
{
  "port": "3010",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "NironeTNode",
      "lnImplementation": "CLT",
      "Authentication": {
        "macaroonPath": "/home/nicolas/c-lightning-REST/certs/",
        "configPath": "/home/nicolas/.lightning/config"
      },
      "Settings": {
        "userPersona": "MERCHANT",
        "themeMode": "DAY",
        "themeColor": "INDIGO",
        "channelBackupPath": "/home/nicolas/backup/node-1",
        "logLevel": "ERROR",
        "lnServerUrl": “https://localhost:3001/v1”
        "fiatConversion": true,
        "currencyUnit": "USD"
      }
    }
  ],
  "multiPassHashed": "a4e36cc1f385e89d27c284930b48d759fc7267bb5e7e1f48b3395084e3df09b0"
}
  GNU nano 4.8                                                  cl-rest-config.json
{
    "PORT": 3001,
    "DOCPORT": 4001,
    "PROTOCOL": "https",
    "EXECMODE": "production",
    "RPCCOMMANDS": ["*"]
}

I’ve checked the port 3001 and it is “open” on the firewall… these are the ports opened

8333                       ALLOW       Anywhere
9737                       ALLOW       Anywhere
9735                       ALLOW       Anywhere
5001                       ALLOW       Anywhere
50002                      ALLOW       Anywhere
3002                       ALLOW       Anywhere
4001                       ALLOW       Anywhere
3003                       ALLOW       Anywhere
3010                       ALLOW       Anywhere
8080                       ALLOW       Anywhere
3001                       ALLOW       Anywhere

I appreciate your support and let’s see where is the cat hidden….
Best

Nicolas Velasco
Gestion – Formación Empresas - Energía

***@***.***
http://www.nironet.es<http://www.nironet.es/>
[twitter]<https://twitter.com/NironetSchool>[sÃ­mbolos-redes-sociales-02-1]<https://www.facebook.com/Nironet>
Nironet SL
Barcelona – Vilanova i la Geltrú
Tel: 938 106 596
Movil: (+34) 607 403 073


De: Suheb ***@***.***>
Enviado el: martes, 14 de septiembre de 2021 18:07
Para: Ride-The-Lightning/RTL ***@***.***>
CC: Nicolas Velasco - NR ***@***.***>; Mention ***@***.***>
Asunto: Re: [Ride-The-Lightning/RTL] ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error" (#808)


Please update this time in the RTL-Config.json file...
"lnServerUrl": https://localhost:3001/v1,
to
"lnServerUrl": "https://localhost:3001/v1",

Also, please upgrade your node version to at least v14.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Ride-The-Lightning/RTL/issues/808#issuecomment-919295511>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVUAE6EWOJQCRTSDFUXINV3UB5XJFANCNFSM5EAHLV3A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
[Atención: E-mail Externo]
//--------------------------------------------------------------------// Por favor, revise nuestra política de protección de datos: Per favor, revisi la nostra política de protecció de dades: https://www.nironet.es/politicas-de-cookies-y-proteccion-de-datos-2 * Antes de imprimir este e-mail, piense en si es realmente necesario: El Medio Ambiente es responsabilidad de todos. * Abans d'imprimir aquest e-mail, pensi en si és realment necessari: El Medi ambient és responsabilitat de tots. * Before printing this e-mail, think if you really need to so. The environment is everyone's responsibility. //--------------------------------------------------------------------//

**Comment by NVG2021** [comment]
Hi,

I was fooling with ports  and I’ve got the RTL server running, but not with your standard info….
***@***.***

So, one step more… I guess I have a service getting the 3001
Best regards

Nicolas Velasco
Gestion – Formación Empresas - Energía

***@***.***
http://www.nironet.es<http://www.nironet.es/>
[twitter]<https://twitter.com/NironetSchool>[sÃ­mbolos-redes-sociales-02-1]<https://www.facebook.com/Nironet>
Nironet SL
Barcelona – Vilanova i la Geltrú
Tel: 938 106 596
Movil: (+34) 607 403 073


De: Suheb ***@***.***>
Enviado el: martes, 14 de septiembre de 2021 18:07
Para: Ride-The-Lightning/RTL ***@***.***>
CC: Nicolas Velasco - NR ***@***.***>; Mention ***@***.***>
Asunto: Re: [Ride-The-Lightning/RTL] ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8080},"error" (#808)


Please update this time in the RTL-Config.json file...
"lnServerUrl": https://localhost:3001/v1,
to
"lnServerUrl": "https://localhost:3001/v1",

Also, please upgrade your node version to at least v14.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Ride-The-Lightning/RTL/issues/808#issuecomment-919295511>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVUAE6EWOJQCRTSDFUXINV3UB5XJFANCNFSM5EAHLV3A>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
[Atención: E-mail Externo]
//--------------------------------------------------------------------// Por favor, revise nuestra política de protección de datos: Per favor, revisi la nostra política de protecció de dades: https://www.nironet.es/politicas-de-cookies-y-proteccion-de-datos-2 * Antes de imprimir este e-mail, piense en si es realmente necesario: El Medio Ambiente es responsabilidad de todos. * Abans d'imprimir aquest e-mail, pensi en si és realment necessari: El Medi ambient és responsabilitat de tots. * Before printing this e-mail, think if you really need to so. The environment is everyone's responsibility. //--------------------------------------------------------------------//

**Comment by ghost** [comment]
Could you elaborate on what you did to resolve the issue? I'm in the same boat.

I'm getting this error:
ERROR: GetInfo => Get Info Error: {"name":"RequestError","message":"Error: connect ECONNREFUSED 127.0.0.1:3002","cause":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"error":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3001},"options":{"url":"https://localhost:3002/v1/getinfo","rejectUnauthorized":false,"json":true,"headers":{},"method":"GET","qs":{},"simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false}}

**RTL-Config.json**

{
  "port": "3000",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "c-lightning Mainnet",
      "lnImplementation": "CLT",
      "Authentication": {
        "macaroonPath": "/home/rtl",
        "configPath": "/data/lightning/config"
      },
      "Settings": {
        "userPersona": "MERCHANT",
        "themeMode": "NIGHT",
        "themeColor": "INDIGO",
        "channelBackupPath": "/data/lightning-backup",
        "logLevel": "ERROR",
        "lnServerUrl": "https://localhost:3001",
        "fiatConversion": true,
        "currencyUnit": "USD"
      }
    }
  ],
  "multiPassHashed": <redacted>
}


**cl-rest-config.json**
{
    "PORT": 3001,
    "DOCPORT": 4001,
    "PROTOCOL": "https",
    "EXECMODE": "production",
    "RPCCOMMANDS": ["*"]
}

Version of RTL - 0.11.2
Version of c-lightning - 0.10.2
Version of bitcoind - 0.22.1
Version of node - 14.18.2

**Comment by NVG2021** [comment]
Hi

Ive i tell you the thruth it was some months ago and short memory....

Let me check, but checking your messsges it looks like there are 2 places for the rtl port assignnment....
On all your texts your config seems to be on port 3000 or 3001, and you get the error on port 3002

Another issue could be if you have a firewall set  and then you have to include the ports 3000, 30001, 3002 to allow the access to them

Let me know and ill figure out more things,as there is one more which i cant recall now

Best
Nicolas

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ghost** [comment]
I was able to resolve my issue. port 4001 was already in use and the config file for RTL isn't referenced when using the plugin.js file. I edited the port in the plugin.js to 4002 and all is good.

**Comment by NVG2021** [comment]
Yes, sorry i could not give all the input.
Now enjoy the sats!!!

Enviado desde mi iPhone

El 3 dic 2021, a las 22:30, nR2Cncdm ***@***.***> escribió:

﻿

I was able to resolve my issue. port 4001 was already in use and the config file for RTL isn't referenced when using the plugin.js file. I edited the port in the plugin.js to 4002 and all is good.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Ride-The-Lightning/RTL/discussions/809#discussioncomment-1746791>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVUAE6CW3VE2QA3AX7PMOBTUPEZHRANCNFSM5JIGVMQQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

[Atención: E-mail Externo]
//--------------------------------------------------------------------// Por favor, revise nuestra política de protección de datos: Per favor, revisi la nostra política de protecció de dades: https://www.nironet.es/politicas-de-cookies-y-proteccion-de-datos-2 * Antes de imprimir este e-mail, piense en si es realmente necesario: El Medio Ambiente es responsabilidad de todos. * Abans d'imprimir aquest e-mail, pensi en si és realment necessari: El Medi ambient és responsabilitat de tots. * Before printing this e-mail, think if you really need to so. The environment is everyone's responsibility. //--------------------------------------------------------------------//
